pkg/agent/run.go

```diff
@@ -114,10 +114,17 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 			return err
 		}
 	}
+
+	notifySocket := os.Getenv("NOTIFY_SOCKET")
+	os.Unsetenv("NOTIFY_SOCKET")
+
 	if err := setupTunnelAndRunAgent(ctx, nodeConfig, cfg, proxy); err != nil {
 		return err
 	}
 
+	os.Setenv("NOTIFY_SOCKET", notifySocket)
+	systemd.SdNotify(true, "READY=1\n")
```

Comment:
Did it not work out to wait until after containerd and kubelet are started?

Reply:
Sadly, no. It stalled out.

Reply:
oh right, I see why.

Reply:
What if we moved this right above `<-ctx.Done()`, after all the other node setup is complete? That seems like it might be a more concrete indication of the node startup being taken care of?